api_v3/services/LiveStreamService.php

```diff
@@ -577,4 +577,102 @@ public function regenerateStreamTokenAction($entryId)
+            {
+                if (in_array($importantCuePointTag, $cuePointTags))
+                {
+                    $containsImportantTag = true;
+                    break;
+                }
+            }
+            if (!$containsImportantTag)
+            {
+                $shouldBeDeletedCuePoints[] = $cuePoint->getId();
+            }
+        }
+
+        CuePointPeer::setCuePointsStatus($shouldBeDeletedCuePoints, CuePointStatus::DELETED);
+
+        $vodEntry = entryPeer::retrieveByPK($liveEntry->getRecordedEntryId());
+        if (!$vodEntry)
+        {
+            KalturaLog::warning("Recorded entry id [" . $liveEntry->getRecordedEntryId() . "] not found");
+        }
+        else {
+            $currentDate = date("M-d-Y H:i");
+            $vodEntry->setName($liveEntry->getName() . " " . $currentDate);
+            $vodEntry->setDescription($liveEntry->getDescription());
+            $vodEntry->setTags($liveEntry->getTags());
+            $vodEntry->setDisplayInSearch(1);
+            $vodEntry->save();
+        }
+
+        $liveEntry->setRecordedEntryId(null);
+        $liveEntry->setRedirectEntryId(null);
+        $liveEntry->save();
+
+        $pushNotificationPlugin = KalturaPluginManager::getPluginInstance(PushNotificationPlugin::getPluginName());
+
+        if ($pushNotificationPlugin)
+        {
+            $liveEntryString = new KalturaStringValue();
+            $liveEntryString->value = $liveEntryId;
+            $pushNotificationParameter = new KalturaPushEventNotificationParameter();
+            $pushNotificationParameter->key = "entryId";
+            $pushNotificationParameter->value = $liveEntryString;
+            $pushEventNotificationParameterArray = new KalturaPushEventNotificationParameterArray();
+            $pushEventNotificationParameterArray[] = $pushNotificationParameter;
+            $pushNotificationParams = new KalturaPushNotificationParams();
+            $pushNotificationParams->userParams = $pushEventNotificationParameterArray;
+
+            $pushNotificationTemplateService = new PushNotificationTemplateService();
+            $pushNotificationTemplateService->initService("pushnotification_pushnotificationtemplate",
+                "eventNotificationTemplate", "sendCommand");
+
+            $pushNotificationTemplateService->sendCommandAction('POLLS_PUSH_NOTIFICATIONS',
+                $pushNotificationParams, KalturaPushNotificationCommandType::CLEAR_QUEUE);
+            $pushNotificationTemplateService->sendCommandAction('PUBLIC_QNA_NOTIFICATIONS',
+                $pushNotificationParams, KalturaPushNotificationCommandType::CLEAR_QUEUE);
+            $pushNotificationTemplateService->sendCommandAction('USER_QNA_NOTIFICATIONS',
+                $pushNotificationParams, KalturaPushNotificationCommandType::CLEAR_QUEUE);
+        }
+        else
+        {
+            KalturaLog::info("Plugin [" . PushNotificationPlugin::getPluginName() . "] is not loaded");
```

Comment:
If plugin not enabled no need to return false.
This method should always return true or throw exception.